Multisig/UI/ClaimToken/EnterCustomAddressViewController.swift

```diff
@@ -0,0 +1,188 @@
+//
+//  EnterCustomAddressViewController.swift
+//  Multisig
+//
+//  Created by Vitaly on 24.07.22.
+//  Copyright © 2022 Gnosis Ltd. All rights reserved.
+//
+
+import UIKit
+import Web3
+
+class EnterCustomAddressViewController: UIViewController {
+
+    @IBOutlet private weak var continueButton: UIButton!
+    @IBOutlet private weak var addressField: AddressField!
+
+    private var stepLabel: UILabel!
+    private var stepNumber: Int = 2
+    private var maxSteps: Int = 3
+
+    private var debounceTimer: Timer!
+    private let debounceDuration: TimeInterval = 0.250
+
+    private var chain: SCGModels.Chain!
+    private var trackingParameters: [String: Any]?
+    private var address: Address? { addressField?.address }
+
+    var mainnet: Bool = true
+    var onContinue: ((_ address: Address) -> ())?
+
+
+    override func viewDidLoad() {
+        super.viewDidLoad()
+
+        ViewControllerFactory.makeTransparentNavigationBar(self)
+        navigationItem.hidesBackButton = false
+        title = "Custom address"
+
+        stepLabel = UILabel(frame: CGRect(x: 0, y: 0, width: 50, height: 21))
+        stepLabel.textAlignment = .right
+        navigationItem.rightBarButtonItem = UIBarButtonItem(customView: stepLabel)
+        stepLabel.setStyle(.tertiary)
+        stepLabel.text = "\(stepNumber) of \(maxSteps)"
+
+        addressField.setPlaceholderText("Enter delegate address")
+        addressField.onTap = { [weak self] in self?.didTapAddressField() }
+
+        continueButton.setText("Select & Continue", .filled)
+        continueButton.isEnabled = false
+
+        if mainnet {
+            chain = SCGModels.Chain.mainnetChain()
+        } else {
+            chain = SCGModels.Chain.rinkebyChain()
+        }
+
+        trackingParameters = { ["chain_id" : chain.chainId.description] }()
+    }
+
+    private func didTapAddressField() {
+        let alertVC = UIAlertController(title: nil, message: nil, preferredStyle: .actionSheet)
```

Comment:
why this is re-implemented? I think we already have a view controller that gets address from the user

Reply:
what view countroller do you mean? Enter safe address screen and address book screen use same approach as here. This could be refactored and extracted to a helper class, but I would do this as a separate task.

Reply:
ticket for refactoring task: https://app.zenhub.com/workspaces/mobile-engineering-60cc78740f488d000e996335/issues/safe-global/safe-ios/2519